admin/config-ui/factories/class-factory-post-type.php

```diff
@@ -23,7 +23,8 @@ public function get_fields() {
 			$post_types = get_post_types( array( 'public' => true ), 'objects' );
 			if ( ! empty( $post_types ) ) {
 				foreach ( $post_types as $post_type => $post_type_object ) {
-					$field = new WPSEO_Config_Field_Choice_Post_Type( $post_type, $post_type_object->label );
+					$label = str_replace( '&#39;', '’', $post_type_object->label );
```

Comment:
I think this shouldn't be the solution. You are now solving one issue, what about other html decoded characters?